octoprint_smsnotifier/__init__.py

```diff
@@ -101,37 +101,34 @@ def _send_txt(self, payload, snapshot=False):
             tonumber = phonenumbers.format_number(phonenumbers.parse(number, 'US'), phonenumbers.PhoneNumberFormat.E164)
-        tags = {
-            'filename': filename,
-            'elapsed_time': elapsed_time,
-            'printer_name': self._settings.get(["printer_name"])
-        }
-        message = self._settings.get(["message_format", "body"]).format(**tags)
-
-        client = TwilioRestClient(self._settings.get(['account_sid']), self._settings.get(['auth_token']))
-        if snapshot:
-            try:
-                client.messages.create(to=tonumber, from_=fromnumber, body=message, media_url=snapshot)
-            except Exception as e:
-                # report problem sending sms
-                self._logger.exception("SMS notification error: %s" % (str(e)))
-                return False
+            tags = {
+                'filename': filename,
+                'elapsed_time': elapsed_time,
+                'printer_name': self._settings.get(["printer_name"])
+            }
+            message = self._settings.get(["message_format", "body"]).format(**tags)
+
+            client = TwilioRestClient(self._settings.get(['account_sid']), self._settings.get(['auth_token']))
+            if snapshot:
+                try:
+                    client.messages.create(to=tonumber, from_=fromnumber, body=message, media_url=snapshot)
+                except Exception as e:
+                    # report problem sending sms
+                    self._logger.exception("SMS notification error: %s" % (str(e)))
+                    return False
+                else:
+                    # report notification was sent
+                    self._logger.info("Print notification sent to %s" % (self._settings.get(['recipient_number'])))
             else:
-                # report notification was sent
-                self._logger.info("Print notification sent to %s" % (self._settings.get(['recipient_number'])))
-                return True
-
-        try:
-            client.messages.create(to=tonumber, from_=fromnumber, body=message)
-        except Exception as e:
-            # report problem sending sms
-            self._logger.exception("SMS notification error: %s" % (str(e)))
-        else:
-            # report notification was sent
-            self._logger.info("Print notification sent to %s" % (self._settings.get(['recipient_number'])))
-            return True
-
-        return False
+                try:
+                    client.messages.create(to=tonumber, from_=fromnumber, body=message)
+                except Exception as e:
+                    # report problem sending sms
+                    self._logger.exception("SMS notification error: %s" % (str(e)))
+                    return False
+                else:
+                    # report notification was sent
+                    self._logger.info("Print notification sent to %s" % (self._settings.get(['recipient_number'])))
```

Comment:
You've removed the "return true" entries on success. If it still returns true it's just a side effect (of self.logger). It appears that `on_event` uses that status (for instance, retrying text-only if sending an image fails).

If it were me, I'd `return true` from the "else" of the try blocks, then `return false` at the very end of the method. That's just a personal preference towards only returning success in known cases.

Reply:
But doesn't `return true` break out of the loop?

Reply:
Oh! Duh, of course. I have an idea but I'm sure you do too.

Reply:
What do you prefer?  Close this pull request and start a new one?  Or close this one and have the contributors fix it?

Reply:
I like what you've done! It's just about how to track and return success or failure, so do it in here. By "I have an idea" I meant "if you get stuck on coding a return to this loop, let me know".

Reply:
@TheKoola @tedder 

I can't find anything in the [docs](https://docs.octoprint.org/en/master/modules/plugin.html?highlight=on_event#octoprint.plugin.types.EventHandlerPlugin) that says we need to return true or false when `on_event()` is called. We can keep it, but I just wouldn't worry about it.